src/updater.h

```diff
@@ -119,7 +131,7 @@ class PollingMetadataUpdater : public MetadataUpdater {
  protected:
   friend class UpdaterTest;
 
-  bool ValidateConfiguration() const;
+  bool ValidateConfiguration() const throw(ValidationError);
```

Comment:
Let's add a new method for "ShouldStartUpdater" or some equivalent name, that returns true or false. If for whatever reason the ShouldStartUpdater "fails" to validate, by not being able to QueryMaster, we should set unhealthy, but then return true so that we make a best effort attempt to poll / watch, as we had already validated that the configuration was valid. This will help keep us safe from transient errors preventing the updaters from firing off. Arguably, this hints at supporting a "SetHealthy", in the event that something is able to recover properly. Let's save that for another PR though.

Reply:
Right, I'll leave this whole discussion in the PR comments for now, as we don't really need any of this functionality yet.